### DISCUSSION TITLE
Get events for devices activated by 3rd-party device

### DISCUSSION THREAD

**Discussion by mveinot** [kuestess/homebridge-platform-insteonlocal#264] [Q&A]
I hope the subject isn't too cryptic - I will try to elaborate:

I was using HomeBridge since before the shutdown and since the shutdown, I converted my old 1st gen hub into a Serial PLM and connected it directly to my HomeBridge VM - for the most part this is working excellently.

There are only a couple exceptions - any device that is typically controlled by a 3rd device - i.e. a Keypad scene, or a motion sensor - basically any event generated by a scene between those devices - the hub doesn't seem to notice. So if I walk into my garage, the motion sensor turns on the ceiling and workbench lights, but HomeBridge and the app still show them as off. If I toggle the light manually at the switch, the status shows as you'd expect.

I think probably the scenes in question don't include the hub - since I probably created them on my 2nd gen hub which I'm not currently using. Is there a way to correct this behaviour - add the hub to the scene or otherwise make it aware of the events? If I can solve this, my cloudless setup will be perfect.

**Comment by mveinot** [comment]
Second, semi-related question: how can I add battery operated devices (like remotes, motion sensors, leak sensors, etc) using the Express interface? I have tried adding them by insteon ID, then pressing the link button on the device, but I haven't really had any success.

**Comment by kuestess** [comment]
@mveinot Your hunch on the first post is correct - you'll need to re-create those scenes on the serial PLM.  The plugin should pick up those events, but only if the PLM is a responder in those scenes.  I've got this situation too (with a PLM and 2245) and just recreated the scenes using the ui.  Works pretty well.  For battery operated devices, unfortunately not a great way to link without some manual interaction on the motion sensor first.  Try this on the motion sensor:
- Hold the set button until the led starts blinking
- Tap the set button (should double blink)
- Link the device via the ui

I think that will get the motion sensor in the right mode to respond to the linking commands.  Let me know how that works.

**Reply by mveinot** [reply]
This is what appears in HomeBridge when I try to link a Motion Sensor II as above: (or any battery device)

Linking 4F.89.7D to hub
Error linking 4F.89.7D to hub
Found Guest Washroom in homebridge config
Found Bathroom Light in homebridge config
Found Garage Light in homebridge config
Found Back Door Lights in homebridge config
Found Workbench Light in homebridge config
Found Deck Lights in homebridge config
Found Downstairs Fan in homebridge config
Found Front Washroom Lights in homebridge config
Found Front Room Plug in homebridge config
Found Pool Pump in homebridge config
Found Basement Plug in homebridge config
Found Bedroom Sconce Right in homebridge config
Found Bedroom Sconce Left in homebridge config
Found Noah's Fan in homebridge config
Saved new insteon config
Created backup of previous config
Done writing Insteon config
Reading devices from insteon.json
TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer or Uint8Array. Received undefined
    at new NodeError (internal/errors.js:322:7)
    at write_ (_http_outgoing.js:722:11)
    at ServerResponse.write (_http_outgoing.js:687:15)
    at InsteonUI.renderLinkPage (/usr/lib/node_modules/homebridge-platform-insteonlocal/insteon-ui.js:1540:6)
    at InsteonUI.handleRequest (/usr/lib/node_modules/homebridge-platform-insteonlocal/insteon-ui.js:1623:8)
    at Layer.handle [as handle_request] (/usr/lib/node_modules/homebridge-platform-insteonlocal/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/usr/lib/node_modules/homebridge-platform-insteonlocal/node_modules/express/lib/router/index.js:328:13)
    at /usr/lib/node_modules/homebridge-platform-insteonlocal/node_modules/express/lib/router/index.js:286:9
    at Function.process_params (/usr/lib/node_modules/homebridge-platform-insteonlocal/node_modules/express/lib/router/index.js:346:12)
    at next (/usr/lib/node_modules/homebridge-platform-insteonlocal/node_modules/express/lib/router/index.js:280:10)
